### PR TITLE
Avoid append when `if_exists` is set to replace

### LINKED ISSUE #1260: dataframe operator is now appending to it's output table instead of replacing.
**Describe the bug**
 the dataframe operator is now appending to it's output table instead of replacing.
refer to https://astronomer.slack.com/archives/C02B8SPT93K/p1668643987896139

**Version**
* Astro: [e.g. 0.6.0]
* OS: [eg. Debian]

**To Reproduce**
1. Write the DAG  '...'
2. Create connection '....'
3. Run using  '....'

**Expected behavior**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by pankajastro**
# Description

closes #1260
## What is the current behavior?
The `load_pandas_dataframe_to_table` does not respect `if_exists` param

## What is the new behavior?
`if_exists` is set to `replace` replace the table rather than append

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/src/astro/databases/snowflake.py [hunk 1]
```diff
@@ -648,8 +648,10 @@ def load_pandas_dataframe_to_table(
         self._assert_not_empty_df(source_dataframe)
 
         auto_create_table = False
-        if if_exists == "replace" or not self.table_exists(target_table):
+        if not self.table_exists(target_table):
```
Comment (sunank200) [inline python-sdk/src/astro/databases/snowflake.py:651]: If the table doesn't exist, don't we require to create a table?

File: python-sdk/src/astro/databases/snowflake.py [hunk 2]
```diff
@@ -648,8 +648,10 @@ def load_pandas_dataframe_to_table(
...
             auto_create_table = True
+        elif if_exists == "replace":
```
Comment (sunank200) [inline python-sdk/src/astro/databases/snowflake.py:653]: What would happen when the table doesn't exist and the user has passed in `replace`?

**Review by sunank200 — COMMENTED** [review]
Added few comments

File: python-sdk/src/astro/databases/snowflake.py (hunk 2 above)
Comment (pankajastro) [reply python-sdk/src/astro/databases/snowflake.py:653]: `auto_create_table` would be true and pandas will take care of creating table

File: python-sdk/src/astro/databases/snowflake.py (hunk 1 above)
Comment (pankajastro) [reply python-sdk/src/astro/databases/snowflake.py:651]: auto_create_table would be true and pandas will take care of creating table

**Review by sunank200 — APPROVED** [review]
LGTM

File: python-sdk/tests/sql/operators/test_dataframe.py [hunk 3]
```diff
@@ -465,3 +465,33 @@ def get_empty_dataframe():
...
+        name="ci_df_replace_table",
+        conn_id=conn_id,
+    )
+    with sample_dag:
+        get_empty_dataframe(output_table=output_tb)
+    test_utils.run_dag(sample_dag)
+    assert output_tb.row_count == 2
+    # re-run dag to and make sure it is replacing table
+    test_utils.run_dag(sample_dag)
+    assert output_tb.row_count == 2
```
Comment (kaxil) [inline python-sdk/tests/sql/operators/test_dataframe.py:497]: This is a brittle test, it checks that there were 2 rows when run multiple times, but it doesn't check that the rows are "replaced"

Comment (pankajastro) [reply python-sdk/tests/sql/operators/test_dataframe.py:497]: hmm, add more tests https://github.com/astronomer/astro-sdk/pull/1293

**Pull request merged**